tests/SpecTests/SearchIndexSpecTest.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+namespace MongoDB\Tests\SpecTests;
+
+use Closure;
+use MongoDB\Collection;
+use MongoDB\Model\CachingIterator;
+use MongoDB\Tests\FunctionalTestCase;
+
+use function bin2hex;
+use function count;
+use function hrtime;
+use function iterator_to_array;
+use function random_bytes;
+use function sleep;
+use function sprintf;
+
+/**
+ * Functional tests for the Atlas Search index management.
+ *
+ * @see https://github.com/mongodb/specifications/blob/master/source/index-management/tests/README.rst#search-index-management-helpers
+ * @group atlas
+ */
+class SearchIndexSpecTest extends FunctionalTestCase
+{
+    private const WAIT_TIMEOUT_SEC = 300;
+    private const STATUS_READY = 'READY';
+
+    public function setUp(): void
+    {
+        if (! self::isAtlas()) {
+            self::markTestSkipped('Search Indexes are only supported on MongoDB Atlas 7.0+');
+        }
+
+        parent::setUp();
+
+        $this->skipIfServerVersion('<', '7.0', 'Search Indexes are only supported on MongoDB Atlas 7.0+');
+    }
+
+    /**
+     * Case 1: Driver can successfully create and list search indexes
+     *
+     * @see https://github.com/mongodb/specifications/blob/master/source/index-management/tests/README.rst#case-1-driver-can-successfully-create-and-list-search-indexes
+     */
+    public function testCreateAndListSearchIndexes(): void
+    {
+        $collection = $this->createCollection($this->getDatabaseName(), $this->getCollectionName());
+        $name = 'test-search-index';
+        $mapping = ['mappings' => ['dynamic' => false]];
+
+        $createdName = $collection->createSearchIndex(
+            $mapping,
+            ['name' => $name, 'comment' => 'Index creation test'],
+        );
+        $this->assertSame($name, $createdName);
+
+        $indexes = $this->waitForIndexes($collection, fn ($indexes) => $this->allIndexesAreQueryable($indexes));
+
+        $this->assertCount(1, $indexes);
+        $this->assertSame($name, $indexes[0]->name);
+        $this->assertDocumentsMatch($mapping, $indexes[0]->latestDefinition);
```

Comment:
Apologies for not specifying, but I was suggesting `assertMatchesDocument()` or `assertSameDocument()` from the base TestCase class instead of comparing JSON strings here.

`assertDocumentsMatch()` is unique to the legacy spec test runner (note that its DocumentsMatchConstraint class differs from the unified test runner's Matches constraint), and I'd be hesitant to move the legacy test runner logic into the main test suite, as the ultimate goal is to delete all of that code once each spec can be ported over to the unified format.

Is it feasible to use one of the two existing TestCase methods here? `assertMatchesDocument()` allows extra fields in the actual document, while `assertSameDocument()` does not -- so you can decide which is most appropriate. If so, you can then revert the introduction of DocumentsMatchConstraint to FunctionalTestCase.